geopandas/array.py

```diff
@@ -405,6 +422,13 @@ def _binary_method(op, left, right, **kwargs):
                     len(left), len(right)
                 )
                 raise ValueError(msg)
+            if not _check_crs(left, right):
+                warnings.warn(
+                    "CRS mismatch between the CRS of left geometries ({0}) "
+                    "and the CRS of right geoemtries ({1}). "
+                    "Use `GeoSeries.to_crs()` to reproject one of "
+                    "the passed geometry arrays.".format(left.crs, right.crs)
```

Comment:
Since the repr of both CRS objects is quite big, I would not include it inside parantheses in the sentence, but maybe put both at the end, something like

Left CRS:\n
{}\n
Right CRS:\n
{}

Reply:
It does not return whole repr, just `epsg:2263` and similar. This is the resulting warning:

```
UserWarning: CRS mismatch between the CRS of left geometries (epsg:2263) and the CRS of right geoemtries (epsg:4326). Use `GeoSeries.to_crs()` to reproject one of the passed geometry arrays.
```

Reply:
Ah, the `__str__` is using the `CRS.srs` and not the full repr, OK.

Now, in most cases that will be something short like EPSG:4326, but depending on how the CRS is created, it can still be long ..

```
In [19]: crs = pyproj.CRS("EPSG:31370")                                                                                                                                                                            

In [20]: str(crs)                                                                                                                                                                                                  
Out[20]: 'EPSG:31370'

In [21]: str(pyproj.CRS(crs.to_wkt()))                                                                                                                                                                             
Out[21]: 'PROJCRS["Belge 1972 / Belgian Lambert 72",BASEGEOGCRS["Belge 1972",DATUM["Reseau National Belge 1972",ELLIPSOID["International 1924",6378388,297,LENGTHUNIT["metre",1]]],PRIMEM["Greenwich",0,ANGLEUNIT["degree",0.0174532925199433]],ID["EPSG",4313]],CONVERSION["Belgian Lambert 72",METHOD["Lambert Conic Conformal (2SP)",ID["EPSG",9802]],PARAMETER["Latitude of false origin",90,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8821]],PARAMETER["Longitude of false origin",4.36748666666667,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8822]],PARAMETER["Latitude of 1st standard parallel",51.1666672333333,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8823]],PARAMETER["Latitude of 2nd standard parallel",49.8333339,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8824]],PARAMETER["Easting at false origin",150000.013,LENGTHUNIT["metre",1],ID["EPSG",8826]],PARAMETER["Northing at false origin",5400088.438,LENGTHUNIT["metre",1],ID["EPSG",8827]]],CS[Cartesian,2],AXIS["easting (X)",east,ORDER[1],LENGTHUNIT["metre",1]],AXIS["northing (Y)",north,ORDER[2],LENGTHUNIT["metre",1]],USAGE[SCOPE["unknown"],AREA["Belgium - onshore"],BBOX[49.5,2.5,51.51,6.4]],ID["EPSG",31370]]'
```

Reply:
Oh, yeah. That is annoying. What about using `crs.to_string()` instead? That gives `EPSG:31370` in both cases of these cases.From docs: _It attempts to convert it to the authority string. Otherwise, it uses the string format of the user input to create the CRS._ 

That will likely minimise the long output. I don't want to print whole specification, it can be annoying to print two CRS repr string.

Reply:
Once you have a WKT that does not match exactly to an EPSG code, to_string will still print the full thing .. 

So I think that is unavoidable that in some (not that special, I think) cases, it will show the full thing. So maybe better putting it at the end of the message then?

Reply:
Okay. I did both, changed it to `crs.to_string()` to get epsg code if possible and moved it to the end if we'll get the full string.

Reply:
Sorry to keep on bothering on this point. 
What I am thinking now is that, if `to_string` does not return a EPSG number but rather the full WKT or proj string or etc, in that case the *repr* might actually be more readable ..

Reply:
Okay, let's go with repr as default, to make it consistent and readable.